drivers/sensor/default_rtio_sensor.c

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright (c) 2023 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/dsp/types.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(rtio_sensor, CONFIG_SENSOR_LOG_LEVEL);
```

Comment:
Might be worth changing this to like sensor_compat or sensor_shim (along with the filename) to really reiterate this is meant to be a compatibility layer with the existing API.